### PR TITLE
Use 64 bits for time_t

### PR DESCRIPTION
**Pull request by sbc100**
This brings us back in line with upstream musl.  The change to 32-bit
was only recently made in #16966. The reason we made this change was
made was because we had certain C library calls that were implemented in
JS that returned `time_t`.  Since returning 64-bit values from JS
functions is not always easy (we don't always have WASM_BIGINT
available) that simplest solution was to define `time_t` to 32-bit which
doesn't have issues at the JS boundary.

However, in the intervening time many of the `time_t`-returning function
have been moved into native code (See #16606 and #16439) with only two
remaining: _mktime_js and _timegm_js.  So this change redefines just
those two functions to return `int` while keeping `time_t` itself as
64-bit.

Fixes: #17393

### CODE REVIEW

File: system/lib/libc/emscripten_time.c [hunk 1]
```diff
@@ -19,8 +19,8 @@ __attribute__((__weak__)) int   daylight = 0;
 __attribute__((__weak__)) char *tzname[2] = { 0, 0 };
 
 void _tzset_js(long* timezone, int* daylight, char** tzname);
-time_t _timegm_js(struct tm *tm);
-time_t _mktime_js(struct tm *tm);
+int _timegm_js(struct tm *tm);
+int _mktime_js(struct tm *tm);
```
Comment (kripken) [inline system/lib/libc/emscripten_time.c:23]: Maybe comment here as to why these are different?

**Review by kripken — COMMENTED** [review]
Should we mention this in the changelog? The type change is user-visible and people might need changes as in those tests changed in this PR?

File: system/lib/libc/emscripten_time.c (hunk 1 above)
Comment (sbc100) [reply system/lib/libc/emscripten_time.c:27]: Done

File: system/lib/libc/emscripten_time.c [hunk 2]
```diff
@@ -19,8 +19,12 @@ __attribute__((__weak__)) int   daylight = 0;
 __attribute__((__weak__)) char *tzname[2] = { 0, 0 };
 
 void _tzset_js(long* timezone, int* daylight, char** tzname);
-time_t _timegm_js(struct tm *tm);
-time_t _mktime_js(struct tm *tm);
+// Declare these functions `int` rather than time_t to avoid int64 at the wasm
+// bounday (avoids 64-bit complexity at the boundary when WASM_BIGINT is
```
Comment (kripken) [inline system/lib/libc/emscripten_time.c:23]:
```suggestion
// boundary (avoids 64-bit complexity at the boundary when WASM_BIGINT is
```

**Review by kripken — APPROVED** [review]

**Comment by hoodmane** [comment]
Thanks!

**Pull request merged**